dll/win32/shell32/folders/CRegFolder.cpp

```diff
@@ -641,26 +641,50 @@ HRESULT WINAPI CRegFolder::GetDisplayNameOf(PCUITEMID_CHILD pidl, DWORD dwFlags,
     }
 
     /* Allocate the buffer for the result */
-    LPWSTR pszPath = (LPWSTR)CoTaskMemAlloc((MAX_PATH + 1) * sizeof(WCHAR));
+    SIZE_T cchPath = MAX_PATH + 1;
+    LPWSTR pszPath = (LPWSTR)CoTaskMemAlloc(cchPath * sizeof(WCHAR));
     if (!pszPath)
         return E_OUTOFMEMORY;
 
     hr = S_OK;
 
     if (GET_SHGDN_FOR (dwFlags) == SHGDN_FORPARSING)
     {
-        wcscpy(pszPath, m_rootPath);
-        PWCHAR pItemName = &pszPath[wcslen(pszPath)];
+        SIZE_T pathlen = 0;
+        PWCHAR pItemName = pszPath; // GET_SHGDN_RELATION(dwFlags) == SHGDN_INFOLDER
+        if (GET_SHGDN_RELATION(dwFlags) != SHGDN_INFOLDER)
+        {
+            hr = StringCchCopyW(pszPath, cchPath, m_rootPath);
+            if (SUCCEEDED(hr))
+            {
+                pathlen = wcslen(pszPath);
+                pItemName = &pszPath[pathlen];
+                if (pathlen)
```

Comment:
In which condition is it possible to have pathlen == 0? If such a thing ever happens, is the code below still constructs a correct path string? (and what happens to the `hr` error code?)

Reply:
It will be 0 if CRegFolder::Initialize is called with an empty string.

Nothing happens to the `hr`, it remains `S_OK`, the only difference is that no `\` is prepended before appending the child pilds name.

It is a corner case but checking if an int is zero is so cheap I think the check should remain.